File: medplum-java/medplum-java-model/src/main/java/com/fhir/SubstanceProtein_Subunit.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Integer;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSubstanceProtein_Subunit.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SubstanceProtein_Subunit {
  Optional<Attachment> sequenceAttachment();

  Optional<Identifier> cTerminalModificationId();

  Optional<String> cTerminalModification();

  Optional<String> sequence();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<Integer> subunit();

  Optional<String> nTerminalModification();

  Optional<Integer> length();

  Optional<String> id();

  Optional<Identifier> nTerminalModificationId();

  static ImmutableSubstanceProtein_Subunit.Builder builder() {
    return ImmutableSubstanceProtein_Subunit.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/SubstanceProtein_Subunit.ts
import { Attachment } from "./Attachment";
import { Extension } from "./Extension";
import { Identifier } from "./Identifier";

export interface SubstanceProtein_Subunit {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    subunit: number | null | undefined;
    sequence: string | null | undefined;
    length: number | null | undefined;
    sequenceAttachment: Attachment | null | undefined;
    nTerminalModificationId: Identifier | null | undefined;
    nTerminalModification: string | null | undefined;
    cTerminalModificationId: Identifier | null | undefined;
    cTerminalModification: string | null | undefined;
}
